docs/style-guide.md

```diff
@@ -0,0 +1,439 @@
+Rust Best Practice for ICU4X
+============================
+
+# Introduction
+
+This document outlines the Rust style guide and best practice for code in ICU4X.
+
+As Rust is a new programming language it is expected that the language guidelines will evolve over time, so care should be taken to keep this document up to date, and any new changes after the initially agreed upon version should be reflected in the change history at the bottom of this document.
+
+Before reading this, it is recommended that you read most of [The Rust Programming Language](https://doc.rust-lang.org/book/title-page.html) and [Rust by Example](https://doc.rust-lang.org/rust-by-example/#rust-by-example). There's enough new concepts and syntax in Rust that it's not a language that's easy to pick up "as you go".
+
+Items in this document are categorized as **required** or **suggested**, which essentially maps to the __MUST__ and __SHOULD__ language of Internet RFCs. Any use of the more lenient MAY is deliberately avoided here, since there's no need to encourage too much personal preference in this code base.
+
+Many of the practices here are collected from existing sources (which should be cited) and where applicable there may be some additional justification for including that practice in this document and why it is categorized as it is. In general where a practice is sensible, expected to be achievable, and has no obvious downside, it will be marked as required.
+
+Note however that none of this document is meant to trump common sense, if if you're in a situation where it would be better to violate a **required** practice, then it just means we should have a discussion about it. There will almost certainly be many cases where we need to update this doc and even go back to existing code to update it in the light of new information. However exceptions should always be commented clearly for the next maintainer.
+
+## Sources
```

Comment:
I feel that this and next section (Other useful links) should go towards the end (where references/tools usually go)

Reply:
True. I guess since this is so unfinished in other ways, I wasn't thinking about adding any kind of appendix just yet. It was originally targeted at an audience who were less Rust savvy, so I wanted to pre-load the document with some pointers.